pros/cli/upload.py

```diff
@@ -83,6 +83,16 @@ def upload(path: Optional[str], project: Optional[c.Project], port: str, **kwarg
         raise dont_send(click.UsageError('No port provided or located. Make sure to specify --target if needed.'))
 
     if kwargs['target'] == 'v5':
+        slot = kwargs.get('slot', None)
+        if (slot == None):
+            try:
+                project_file = open(str(project.location)+"/project.pros", 'r')
+                import json
+                project_file_json = json.loads(project_file.read())
+                kwargs['slot'] = project_file_json['py/state']['upload_options']['slot']
+            except Exception:
+                kwargs['slot'] = 1
```

Comment:
```suggestion
        if slot is None and project is not None:
            slot = project.upload_options.get('slot', None)
        if slot is None:
            slot = 1
```
or something to that effect. No need to re-parse the project file, it's already been parsed.